packages/core/src/pubsub/incoming-channel.ts

```diff
@@ -16,89 +16,69 @@ export type IPFSPubsubMessage = {
   key: Uint8Array
 }
 
-function buildResubscribeQueue(logger: DiagnosticsLogger) {
-  return new TaskQueue((error, retry) => {
-    if (error.message.includes('Already subscribed')) {
-      logger.debug(error.message)
-    } else if (error.message.includes('The user aborted a request')) {
-      // For some reason the first call to pubsub.subscribe throws this error.
-      retry()
-    } else {
-      logger.err(error)
-    }
-  })
-}
-
 /**
- * Incoming IPFS PubSub message stream as Observable.
- * Ensures that ipfs subscription to a topic is live by periodically re-subscribing.
- *
- * One could simplify this by removing periodic re-subscription and serialization via task-queue.
- * Unfortunately, we can end up with concurrency issues: trying to unsubscribe while (or right after) other instance tries to subscribe.
- * So, better keep it all together.
+ * Subscription attempts must be sequential, in FIFO order.
```

Comment:
why is it important that subscription attempts be serialized?  What happens if they are not?

Reply:
".subscribe" call is not idempotent. It throws if already subscribed: https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-http-client/src/pubsub/subscription-tracker.js#L26